### ISSUE TITLE
 the terminal is getting "stuck" on signing you into the 2nd Partner account at the CLI..

### ISSUE THREAD

**Issue by protolabzmanpreetphp** [Shopify/cli#3625]
### Please confirm that you have:

- [X] Searched [existing issues](.) to see if your issue is a duplicate. (If you’ve found a duplicate issue, feel free to add additional information in a comment on it.)
- [X] Reproduced the issue in [the latest CLI version](https://www.npmjs.com/package/@shopify/cli).

### In which of these areas are you experiencing a problem?

Extension

### Expected behavior

Previously, I worked on a different theme app extension and a different organization. After switching to another theme app extension and another organization, when I push the code to their partner organization, it always shows previous details even after logging out of Shopify from the terminal, which runs successfully with the new login as well.

### Actual behavior

Need to shown and push the code to their partner organization details.

### Verbose output

```shell
I am using this command : npm run dev -- --reset
```

### Reproduction steps

1. Changed to the app directory of theme app extension
2. run :  npm run dev
3. Also run : npm run dev -- --reset

### Operating System

Windows 10 Pro, 64-bit operating system, ubuntu 20.04

### Shopify CLI version (check your project's `package.json` if you're not sure)

3.0

### Shell

cmd

### Node version (run `node -v` if you're not sure)

v18.15.0

### What language and version are you using in your application?

Ruby 3.2.2

**Comment by amcaplan** [comment]
The CLI maintains state for `dev` and `deploy` separately. If you want to deploy to a different app, you'll need to run `npm run deploy -- --reset`.

Does that clear things up?

**Comment by protolabzmanpreetphp** [comment]
yes, I did this but it not works.

**Comment by amcaplan** [comment]
@protolabzmanpreetphp can you please give fuller details about what you tried and what happened?

**Comment by protolabzmanpreetphp** [comment]
I worked on a different theme app extension and a different partner account org. After switching to another theme app extension and another org, when I push the code to their partner organization, it always shows previous details of partner account org & even after logging out of Shopify from the terminal.

**Comment by protolabzmanpreetphp** [comment]
The current login organization on the terminal is not displayed when code is pushed.

**Comment by amcaplan** [comment]
Logging out will refresh your personal credentials, but it will not change how the CLI associates your app with an app in Partners.

When you run `npm run deploy -- --reset` you should be prompted to choose an organization and app.

As I said before, `dev` and `deploy` each maintain their own state, so you have to use `--reset` on `deploy` if you want it to affect deploys.

**Comment by protolabzmanpreetphp** [comment]
I tried with both dev and deploy, but It not works.

**Comment by isaacroldan** [comment]
Hi @protolabzmanpreetphp, we'd need more detailed steps on what you are trying. 
Are you changing to a different folder? or always working on the same app project? what did `--reset` do and why didn't it work?

**Comment by protolabzmanpreetphp** [comment]
No, I am working on a different project and its associated partner account. I also did Shopify logout and login with another partner account, but it always shows the details of the previous partner account.

**Comment by isaacroldan** [comment]
@protolabzmanpreetphp, as I said, we'd need a more detailed step by step explanation on what you are doing. 
You might have cached info in your `toml` file from the other account, but it's hard to know. (send us step-by-step, and/or verbose logs)

If you want to share more private details (like which orgs/apps) are you connecting too my recommendation would be to open a support ticket with those details.

The CLI is independent from folder to folder, so there is no way to show previous info unless you have something cached in your project.

**Comment by protolabzmanpreetphp** [comment]
please check this video, https://www.awesomescreenshot.com/video/26390543?key=d71ab5723b0d2ca62fcfe07a32a0d79f 
then you can understand the issue with all steps..

**Comment by isaacroldan** [comment]
You are mixing two different CLIs, see the warnings when you run `shopify logout`, it warns you that CLI 2.x is deprecated.

You have CLI 2.x installed globally in your system and CLI 3.x locally in your project.
When you run `shopify logout` or `shopify login` you are working with the old one, when you run `npm run dev` you are using the new one (3.x)

To properly logout in CLI 3.x run: 
```bash
npm run shopify auth logout 
```

I'd recommend uninstalling the global shopify (version 2.x) to avoid confusion.

We are working on a new version of the Global CLi, but that's not ready yet.

**Comment by protolabzmanpreetphp** [comment]
Its works, Thank you @isaacroldan !!
